examples/Nat/ITransportAddress.cs

```diff
@@ -3,18 +3,57 @@
 
 namespace Pax.Examples.Nat
 {
+  /// <summary>
+  /// Provides typed access to Transport layer addresses, such as TCP ports.
+  /// </summary>
+  /// <typeparam name="T">The type of packet.</typeparam>
   internal interface ITransportAddress<T> : IEquatable<ITransportAddress<T>> where T : Packet
   {
+    /// <summary>
+    /// Determines if the packet claims to originate from this Transport-layer address.
+    /// </summary>
+    /// <param name="packet">The packet to check</param>
+    /// <returns>True if the packet claims to come from this address.</returns>
     bool IsSourceOf(T packet);
+
+    /// <summary>
+    /// Rewrites the Transport-layer source address of the packet to this address.
+    /// </summary>
+    /// <param name="packet">The packet to rewrite</param>
     void SetAsSourceOf(T packet);
+
+    /// <summary>
+    /// Determines if the packet is intended for this Transport-layer address.
+    /// </summary>
+    /// <param name="packet">The packet to check</param>
+    /// <returns>True if the packet is intended for this address.</returns>
```

Comment:
Am I correct in thinking that "this address" is the IP address of the NAT on the network port on which the packet was received?